### ISSUE TITLE
Automate cleanup of any Helix agent log files older than 30 days and offline agent files for non-heartbeating agents

### ISSUE THREAD

**Issue by MattGal** [dotnet/arcade#8663]
We keep a rolling 100 MB collection of logs for every helix agent today; once 100 MB is hit, we delete the oldest 10 MB segment and keep going.

The problem lies in two things, mainly Virtual machines (which have ever-growing, distinct machine names) and docker scenarios (which have random machine names).  We've accumulated a LOT of storage usage in this format of logs, and many tiny blobs for offlined, long gone machines

There should be a regular webjob (or jobs) on the Helix service fabric instance that patrol the agent logs and offline blobs and delete anything aged-out; this will reduce storage costs as well as make these accounts far more "tool-browse-able"

**Comment by ChadNedzlek** [comment]
There is something called [Lifecycle management policies](https://docs.microsoft.com/en-us/azure/storage/blobs/lifecycle-management-overview?tabs=azure-portal) that might be able to do this automatically? That'd be cool, and we could use it for the job storage too, since it ALSO expires.

**Comment by ChadNedzlek** [comment]
Oooh, that's exactly what we want.

**Comment by MattGal** [comment]
Lifecycle management policies do indeed seem to be exactly what we want, and I've prototyped setting them for the offline / job cancelled blobs ("execution" container in helixosautostaging) and the agent-logs container in helixagentlogsint.  

Once I got the right support libraries, it was a satisfyingly small amount of code to write.  Unfortunately these policies actually only _run_ once a day, so I've set them and will check back tomorrow to see what happened.  In the mean time I'll be looking at how to set these policies programmatically during our existing deployments.
